Modules/Numerics/Optimizers/include/itkRegularStepGradientDescentBaseOptimizer.h

```diff
@@ -74,23 +72,16 @@ class ITKOptimizers_EXPORT RegularStepGradientDescentBaseOptimizer : public Sing
   /** Run-time type information (and related methods). */
   itkTypeMacro(RegularStepGradientDescentBaseOptimizer, SingleValuedNonLinearOptimizer);
 
-  using StopConditionRegularStepGradientDescentBaseOptimizerEnum =
-    RegularStepGradientDescentBaseOptimizerEnums::StopConditionRegularStepGradientDescentBaseOptimizer;
+  using StopConditionEnum = RegularStepGradientDescentBaseOptimizerEnums::StopCondition;
 #if !defined(ITK_LEGACY_REMOVE)
   // We need to expose the enum values at the class level
   // for backwards compatibility
-  static constexpr StopConditionRegularStepGradientDescentBaseOptimizerEnum GradientMagnitudeTolerance =
-    StopConditionRegularStepGradientDescentBaseOptimizerEnum::GradientMagnitudeTolerance;
-  static constexpr StopConditionRegularStepGradientDescentBaseOptimizerEnum StepTooSmall =
-    StopConditionRegularStepGradientDescentBaseOptimizerEnum::StepTooSmall;
-  static constexpr StopConditionRegularStepGradientDescentBaseOptimizerEnum ImageNotAvailable =
-    StopConditionRegularStepGradientDescentBaseOptimizerEnum::ImageNotAvailable;
-  static constexpr StopConditionRegularStepGradientDescentBaseOptimizerEnum CostFunctionError =
-    StopConditionRegularStepGradientDescentBaseOptimizerEnum::CostFunctionError;
-  static constexpr StopConditionRegularStepGradientDescentBaseOptimizerEnum MaximumNumberOfIterations =
-    StopConditionRegularStepGradientDescentBaseOptimizerEnum::MaximumNumberOfIterations;
-  static constexpr StopConditionRegularStepGradientDescentBaseOptimizerEnum Unknown =
-    StopConditionRegularStepGradientDescentBaseOptimizerEnum::Unknown;
+  static constexpr StopConditionEnum GradientMagnitudeTolerance = StopConditionEnum::GradientMagnitudeTolerance;
```

Comment:
Shouldn't we also have `using StopConditionRegularStepGradientDescentBaseOptimizer = StopCondition;`?